packages/typespec-client-generator-core/design-docs/client.md

```diff
@@ -0,0 +1,379 @@
+1. Endpoint parameter: it is converted from `@server` definition on the service the client belong to.
+2. Credential parameter: it is converted from `@useAuth` definition on the service the client belong to.
+3. API version parameter: if the service is versioned, then the API version parameter on method will be elevated to client.
+4. Subscription ID parameter: if the service is an ARM service, then the subscription ID parameter on method will be elevated to client.
+
+The `SdkInitializationType` has `access` properties. Currently, the value for clients is `public` while the value for sub clients is `internal`.
+
+### Customization for initialization/accessor parameters
+TCGC has `@clientInitialization` and `@paramAlias` to do the customization for the client initialization/accessor parameters. Users could elevate any method's parameter to the clients.
+
+For example:
+```typespec
+@service({
+  title: "My Service",
+})
+namespace MyService {
+  interface InnerGroup {
+    op upload(@path blobName: string): void;
+  }
+}
+
+namespace MyCustomizations {
+  model InnerGroupClientOptions {
+    blob: string;
+  }
+
+  @@clientInitialization(InnerGroup, InnerGroupClientOptions);
+  @@paramAlias(InnerGroupClientOptions.blob, "blobName");
+}
+```
+The above tsp gets client `MyServiceClient` and sub client `InnerGroup`. 
+The `MyServiceClient` has a client accessor method. The method has a parameter named `blob`. 
+The `InnerGroup`'s `initialization` model's properties contains a property named `blob`. 
+The method `upload` no longer has `blobName` parameter, its corresponding operation's parameter `blobName` is mapped to the client `blob` parameter.
+
+### Problem with current implementation
+With current implementation, we could support user scenario 1 and 2.
+
+1. Current implementation does not have a way to set whether we could initialize the sub client directly. 
+The initialization for client is always `public`, while sub client `internal`. 
+2. The client accessor method is always generated and set to `internal`.
+There is no way to indicate that a sub client could not be initialized from the parent client.
+3. `@operationGroup` is an Azure concept, which means a sub client that could not be initialized directly. 
+It is better to consolidate `@client` and `@operationGroup`.
+
+## New proposal
+
+1. Change `@clientInitialization` decorator and add `clientInitialization` property to `SdkClientType`
+- Change `@clientInitialization` decorator's `options` parameter to `ClientInitializationOptions` type to accept `access` and `accessorAccess` setting.
+- Add `clientInitialization` property to `SdkClientType`. It indicate the initialization method's signature, including the parameter list and access.
+- When `access` set to `Access.public`, it means this client could be initialized separately, while `Access.internal` could not. Default to `Access.public` for client and `Access.internal` for sub client.
```

Comment:
can you make clear that this is the `access` for the initialization property?

With this design, there seem to be 3 accesses associated with a `client`.
1. The access of the `client` itself. This can't be toggled rn, it's always `public`
2. The access of the client's init. If `public`, can directly init, else you can not
3. The access of the client's accessor. If `public`, you can create this client from its parent. Else, not